src/main/g8/src/main/scala/com/example/Main.scala

```diff
@@ -42,11 +43,11 @@ object Main extends IOApp.Simple {
         .withPort(thePort)
         .withHost(theHost)
         .withHttpApp(routes.orNotFound)
+        .withShutdownTimeout(1.second)
         .build
         .productL(IO.println(message).toResource)
     }
-    .useForever
-    .race(IO.readLine)
+    .surround(IO.readLine)
```

Comment:
thought: slight change of semantics, this makes it so that you can only cancel with a readLine once the resource has been allocated. Previously, if the resource is cancelable, you could prevent the server from even starting. I'm not sure if it matters for Ember though.